### PR TITLE
Run all program-test in SBF mode and speed up CI

### PR DESCRIPTION
**Pull request by buffalu**
- Run in SBF mode to make sure actual BPF and runtime restrictions are being enforced. This should help us catch runtime errors before they make it to prod.
- Speed up CI to get fast iteration cycles
- Cleanup some compiler warnings by renaming functions to not have overlap

Screenshot 2024-02-07 at 11 48 27 PM

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -3,97 +3,149 @@ on:
...
+      - uses: actions-rust-lang/setup-rust-toolchain@v1
+      - name: Install solana-verify from crates.io
+        uses: baptiste0928/cargo-install@v3
+        with:
+          crate: solana-verify
+      - name: Install anchor-cli from crates.io
+        uses: baptiste0928/cargo-install@v3
+        with:
+          crate: anchor-cli
+          version: "0.29.0"
```
Comment (ebatsell) [inline .github/workflows/build.yaml:77]: this should be anchor 0.28

File: programs/validator-history/idl/validator_history.json [hunk 2]
```diff
@@ -797,7 +797,7 @@
                 "type": "u8"
               },
               {
-                "name": "num_addrs",
+                "name": "numAddrs",
```
Comment (ebatsell) [inline programs/validator-history/idl/validator_history.json:800]: this shouldn't change as long as we're still on anchor 0.28, maybe the solana program test compatibility is fixed with the new way we're loading in programs but we'd still need to bump versions and make the code compatible

**Review by ebatsell — COMMENTED** [review]
so what's the new command to run tests? the `test` script you had earlier is missing

File: .github/workflows/build.yaml (hunk 1 above)
Comment (buffalu) [reply .github/workflows/build.yaml:77]: CI not liking 0.28.0. unless you feel strongly about it, 0.29.0 seems fine as 0.28.0 never made it to crates.io

**Comment by buffalu** [comment]
> so what's the new command to run tests? the `test` script you had earlier is missing

added to readme

**Review by ebatsell — APPROVED** [review]

**Pull request merged**